extensions/amp-form/0.1/amp-form.js

```diff
@@ -177,8 +185,69 @@ export class AmpForm {
 
 
 /**
+ * Reports validity for the first invalid input - if any.
+ * @param {!HTMLFormElement} form
+ * @private
+ */
+function reportFormValidity(form) {
+  const inputs = toArray(form.querySelectorAll('input,select,textarea'));
```

Comment:
@dvoytenko for moving this to `dom.js` When `:scope` is not available do we jsut do `querySelectorAll` or did you want us to navigate the DOM and check child by child (and its children and so on)?